hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/scale/ITestS3AConcurrentOps.java

```diff
@@ -89,38 +111,48 @@ public void setup() throws Exception {
       }
     }
     LOG.info("Data generated...");
-  }
-
-  private S3AFileSystem getRestrictedFileSystem() throws Exception {
-    Configuration conf = getConfiguration();
-    conf.setInt(MAX_THREADS, 2);
-    conf.setInt(MAX_TOTAL_TASKS, 1);
 
-    conf.set(MIN_MULTIPART_THRESHOLD, "10M");
-    conf.set(MULTIPART_SIZE, "5M");
-
-    S3AFileSystem s3a = getFileSystem();
-    URI rootURI = new URI(conf.get(TEST_FS_S3A_NAME));
-    s3a.initialize(rootURI, conf);
-    return s3a;
-  }
-
-  private S3AFileSystem getNormalFileSystem() throws Exception {
-    S3AFileSystem s3a = new S3AFileSystem();
-    Configuration conf = createScaleConfiguration();
-    URI rootURI = new URI(conf.get(TEST_FS_S3A_NAME));
-    s3a.initialize(rootURI, conf);
-    return s3a;
-  }
+    ExecutorService executor = Executors.newFixedThreadPool(
+        concurrentRenames, new ThreadFactory() {
+          private AtomicInteger count = new AtomicInteger(0);
 
-  @After
-  public void teardown() throws Exception {
-    super.teardown();
-    if (auxFs != null) {
-      auxFs.delete(testRoot, true);
+          public Thread newThread(Runnable r) {
+            return new Thread(r,
+                "testParallelRename" + count.getAndIncrement());
+          }
+        });
+    try {
+      ((ThreadPoolExecutor)executor).prestartAllCoreThreads();
+      Future<Boolean>[] futures = new Future[concurrentRenames];
+      for (int i = 0; i < concurrentRenames; i++) {
+        final int index = i;
+        futures[i] = executor.submit(new Callable<Boolean>() {
+          @Override
+          public Boolean call() throws Exception {
+            NanoTimer timer = new NanoTimer();
+            boolean result = fs.rename(source[index], target[index]);
```

Comment:
Have a look at how DurationInfo works/is used; you can hand off the collection and logging to it